torch/_dynamo/variables/functions.py

```diff
@@ -1510,8 +1521,13 @@ def call_function(
         args = ()
 
         if "async_op" in kwargs and kwargs["async_op"].as_python_constant():
-            unimplemented(
-                f"CollectiveFunctionRewriteVariable can't support async_op=True for {self.fn}"
+            unimplemented_v2(
+                gb_type="async_op=True for distributed collectives",
+                context=f"{self.fn}",
+                explanation="`torch.compile` currently can't trace these",
+                hints=[
+                    *graph_break_hints.DIFFICULT,
```

Comment:
Is this supportable?

Reply:
Not sure, let's just say it is to encourage more user action. Updated.